pkg/controller/servicebindingrequest/nested/accumulator/accumulator.go

```diff
@@ -0,0 +1,44 @@
+package accumulator
+
+import (
+	"errors"
+
+	"github.com/imdario/mergo"
+)
+
+const valuesKey = "values"
+
+// Accumulator is a value accumulator.
+type Accumulator map[string]interface{}
+
+// UnsupportedTypeErr is returned when an unsupported type is encountered.
+var UnsupportedTypeErr = errors.New("unsupported type")
+
+// Accumulate accumulates the `val` value. An error is returned in the case
+// `val` contains an unsupported type.
+func (a Accumulator) Accumulate(val interface{}) error {
+	b := NewAccumulator()
+	switch v := val.(type) {
+	case map[string]interface{}:
+		b[valuesKey] = []map[string]interface{}{v}
+	case string:
+		b[valuesKey] = []string{v}
+	case int:
+		b[valuesKey] = []int{v}
+	case []map[string]interface{}, []string, []int:
+		b[valuesKey] = v
+	default:
+		return UnsupportedTypeErr
```

Comment:
Instead of just returning "unsupported type", it be good to also mention what  is unsupported. In this case, it would be `val`.